src/resources/localDatabase.ts

```diff
@@ -0,0 +1,83 @@
+import { spawnSync } from 'child_process';
+
+const port = 3306;
+const databaseName = 'CVSBNOP';
+const containerName = 'mysql-test-jenkins';
+
+function runCommand(command: string, options: string[]): string {
+  const output = spawnSync(command, options);
+
+  if (String(output.stderr).length > 0) {
+    throw new Error(`Error initialising database: ${String(output.stderr)}`);
+  }
+
+  return String(output.stdout);
+}
+
+function containerExistsAndIsRunning() {
+  const response = runCommand('docker', ['ps', '-a']);
+  const lines = response.split('\n');
+  const containerLine = lines.filter((line) => line.toLowerCase().includes(containerName));
+
+  if (containerLine.length === 0) {
+    return { exists: false, isRunning: false };
+  }
+
+  if (containerLine[0].toLowerCase().includes('exited')) {
+    return { exists: true, isRunning: false };
+  }
+
+  return { exists: true, isRunning: true };
+}
+
+function startDb() {
+  spawnSync('docker', ['start', containerName]);
+}
+
+function createDb() {
+  const dockerResult = runCommand('docker', [
+    'run',
+    '--name',
+    'mysql-test-jenkins',
+    '-p',
+    `${port}:${port}`,
+    '-e',
+    `MYSQL_DATABASE=${databaseName}`,
+    '-e',
+    'MYSQL_ROOT_PASSWORD=12345',
+    '-d',
+    'mysql:5.7',
+  ]);
+  console.log(dockerResult);
+  setTimeout(() => {
```

Comment:
Why are setting a timeout here?

Reply:
To allow the docker container to come up and the DB to be available before we try to put stuff into it. The command above will exit before the DB exists to be written to, so this seemed like the easiest way around it, given that you only need to start it up once per day-ish